File: session-02-dependency-injection/typescript/src/order/order.service.ts
import {Injectable} from "@nestjs/common";
import {Order} from "./order.model";
import {OrderRepository} from "./order.repository";

@Injectable()
export class OrderService {
    // Constructor injection: NestJS reads the type metadata (OrderRepository)
    // and injects the matching provider from the module.
    constructor(private readonly repository: OrderRepository) {
    }

    listOrders(): Order[] {
        return this.repository.findAll();
    }

    findOrder(id: number): Order | undefined {
        return this.repository.findById(id);
    }
}

File: session-02-dependency-injection/java/src/main/java/coffeeshop/order/OrderService.java
package coffeeshop.order;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class OrderService {

    // Constructor injection: Spring sees this constructor, finds an OrderRepository
    // bean in the registry, and passes it here. No @Autowired needed (since Spring 4.3).
    private final OrderRepository repository;

    public OrderService(OrderRepository repository) {
        this.repository = repository;
    }

    public List<Order> listOrders() {
        return repository.findAll();
    }

    public Optional<Order> findOrder(int id) {
        return repository.findById(id);
    }
}
